src/Common/src/TypeSystem/IL/ILProvider.cs

```diff
@@ -185,6 +185,16 @@ private MethodIL CreateMethodIL(MethodDesc method)
                 return new InstantiatedMethodIL(method, methodDefinitionIL);
             }
             else
+            if (method is DelegateMarshallingMethodThunk)
```

Comment:
Can you factor the classes in a way where the `ILStubMethod.EmitIL` override just does the right thing for these? `ILProvider` really shouldn't know about the existence of these types and shouldn't special case them.